pkg/crc/constants/constants.go

```diff
@@ -119,9 +119,14 @@ func GetHomeDir() string {
 	return homeDir
 }
 
-// EnsureBaseDirectoryExists create the ~/.crc directory if it is not present
+// EnsureBaseDirectoriesExist creates ~/.crc, ~/.crc/bin and ~/.crc/cache directories if it is not present
 func EnsureBaseDirectoriesExist() error {
-	return os.MkdirAll(CrcBaseDir, 0750)
+	var err error
+	baseDirectories := []string{CrcBaseDir, MachineCacheDir, crcBinDir}
+	for _, baseDir := range baseDirectories {
+		err = os.MkdirAll(baseDir, 0750)
+	}
+	return err
```

Comment:
`err` does not seem very useful. This is ignoring all errors except the last one, and this returns nil if the last Mkdir is successful. Why is the last error special compared to the others?

Reply:
When an installer is used, `crcBinDir` will be in a location where the user can't write. The directory already exists, so I don't think this will cause a problem, but did you test this scenario?

Reply:
@cfergeau I tested this Patch with installer and it works now. `crcBinDir` is used for identify other executable and here we are using `CrcSymlinkPath = filepath.Join(crcBinDir, "crc")`  which is not depend on `BinDir()` function.